wgpu/src/backend/webgpu.rs

```diff
@@ -1417,83 +1523,176 @@ impl crate::context::Context for ContextWebGpu {
+                        })
+                        .map_err(|err| crate::CompilationInfo {
+                            messages: vec![crate::CompilationMessage {
+                                message: err.to_string(),
+                                message_type: crate::CompilationMessageType::Error,
+                                span: None,
+                            }],
+                        })?;
+                    validate_transformed_shader_module(&spv_module, "", desc).map(|v| {
+                        (
+                            v,
+                            WebShaderCompilationInfo::Transformed {
+                                compilation_info: CompilationInfo { messages: vec![] },
+                            },
+                        )
+                    })
+                }
+                #[cfg(feature = "glsl")]
+                crate::ShaderSource::Glsl {
+                    ref shader,
                     stage,
-                    defines: defines.clone(),
-                };
-                let mut parser = front::glsl::Frontend::default();
-                let glsl_module = parser.parse(&options, shader).unwrap();
-
-                let mut validator = valid::Validator::new(
-                    valid::ValidationFlags::all(),
-                    valid::Capabilities::all(),
-                );
-                let glsl_module_info = validator.validate(&glsl_module).unwrap();
-
-                let writer_flags = naga::back::wgsl::WriterFlags::empty();
-                let wgsl_text =
-                    back::wgsl::write_string(&glsl_module, &glsl_module_info, writer_flags)
-                        .unwrap();
-                webgpu_sys::GpuShaderModuleDescriptor::new(wgsl_text.as_str())
+                    ref defines,
+                } => {
+                    use naga::front;
+
+                    // Parse the given shader code and store its representation.
+                    let options = front::glsl::Options {
+                        stage,
+                        defines: defines.clone(),
+                    };
+                    let mut parser = front::glsl::Frontend::default();
+                    let glsl_module = parser
+                        .parse(&options, shader)
+                        .map_err(|inner| wgc::pipeline::ShaderError {
+                            source: shader.to_string(),
+                            label: desc.label.map(|s| s.to_string()),
+                            inner: Box::new(inner),
+                        })
+                        .map_err(|err| crate::CompilationInfo {
+                            messages: vec![crate::CompilationMessage {
+                                message: err.to_string(),
+                                message_type: crate::CompilationMessageType::Error,
+                                span: err
+                                    .inner
+                                    .errors
```

Comment:
err.to_string just concats the strings for all underlying glsl parsing errors if I follow it correctly. So why not do `err.inner.errors.map` and map each error to a `CompilationMessage`?

Reply:
True, good point. It feels a tad inconsistent that the glsl error type has multiple underlying errors while the other error types only seem to have one.
Should anything be done about that?